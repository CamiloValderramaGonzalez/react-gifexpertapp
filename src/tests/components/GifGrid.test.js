import React from 'react'
import { shallow } from "enzyme";
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el componente GifGrid', () => {

    test('Debe mostrarse correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category='One Punch' />)

        expect(wrapper).toMatchSnapshot();
    })

    test('Debe mostrar items', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://url.com',
            title: 'titulo'
        },
        {
            id: 'ABCD',
            url: 'http://url.com',
            title: 'titulo2'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category='One Punch' />)

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    })
})