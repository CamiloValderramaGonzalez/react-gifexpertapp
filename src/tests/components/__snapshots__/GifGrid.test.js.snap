// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente GifGrid Debe mostrar items 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="titulo"
      url="http://url.com"
    />
    <GifGridItem
      id="ABCD"
      key="ABCD"
      title="titulo2"
      url="http://url.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente GifGrid Debe mostrarse correctamente 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  <p>
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
